cloud/scope/powervs_cluster_test.go

```diff
@@ -1704,3 +1704,521 @@ func TestCreateServiceInstance(t *testing.T) {
+		},
+		{
+			name: "Resource type is invalid",
+			clusterScope: PowerVSClusterScope{
+				IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{},
+			},
+			expectedName: nil,
+		},
+	}
+	for _, tc := range testCases {
+		g := NewWithT(t)
+		t.Run(tc.name, func(_ *testing.T) {
+			rgID := tc.clusterScope.GetServiceName(tc.resourceType)
+			g.Expect(rgID).To(Equal(tc.expectedName))
+		})
+	}
+}
+
+func TestGetVPCByName(t *testing.T) {
+	var (
+		mockVPC  *mock.MockVpc
+		mockCtrl *gomock.Controller
+	)
+
+	setup := func(t *testing.T) {
+		t.Helper()
+		mockCtrl = gomock.NewController(t)
+		mockVPC = mock.NewMockVpc(mockCtrl)
+	}
+	teardown := func() {
+		mockCtrl.Finish()
+	}
+	t.Run("When GetVPCByName returns error", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScope := PowerVSClusterScope{
+			IBMVPCClient:      mockVPC,
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{ObjectMeta: metav1.ObjectMeta{Name: "ClusterName"}},
+		}
+		mockVPC.EXPECT().GetVPCByName(gomock.Any()).Return(nil, fmt.Errorf("GetVPCByName returns error"))
+		vpcResponse, err := clusterScope.getVPCByName()
+		g.Expect(err).ToNot(BeNil())
+		g.Expect(vpcResponse).To(BeNil())
+	})
+	t.Run("When GetVPCByName returns valid vpc details", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScope := PowerVSClusterScope{
+			IBMVPCClient:      mockVPC,
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{ObjectMeta: metav1.ObjectMeta{Name: "ClusterName"}},
+		}
+		vpcOutput := &vpcv1.VPC{Name: ptr.To("VPCName"), ID: ptr.To("vpcID"), DefaultSecurityGroup: &vpcv1.SecurityGroupReference{ID: ptr.To("DefaultSecurityGroupID")}}
+		mockVPC.EXPECT().GetVPCByName(gomock.Any()).Return(vpcOutput, nil)
+
+		vpcResponse, err := clusterScope.getVPCByName()
+		g.Expect(err).To(BeNil())
+		g.Expect(vpcResponse.ID).To(Equal(vpcOutput.ID))
+	})
+}
+
+func TestCheckVPC(t *testing.T) {
+	var (
+		mockVPC  *mock.MockVpc
+		mockCtrl *gomock.Controller
+	)
+
+	setup := func(t *testing.T) {
+		t.Helper()
+		mockCtrl = gomock.NewController(t)
+		mockVPC = mock.NewMockVpc(mockCtrl)
+	}
+	teardown := func() {
+		mockCtrl.Finish()
+	}
+	t.Run("When GetVPC returns success", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScope := PowerVSClusterScope{
+			IBMVPCClient:      mockVPC,
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{Spec: infrav1beta2.IBMPowerVSClusterSpec{VPC: &infrav1beta2.VPCResourceReference{ID: ptr.To("VPCID")}}},
+		}
+		vpcOutput := &vpcv1.VPC{Name: ptr.To("VPCName"), ID: ptr.To("VPCID"), DefaultSecurityGroup: &vpcv1.SecurityGroupReference{ID: ptr.To("DefaultSecurityGroupID")}}
+		mockVPC.EXPECT().GetVPC(gomock.Any()).Return(vpcOutput, nil, nil)
+		vpcID, err := clusterScope.checkVPC()
+		g.Expect(err).To(BeNil())
+		g.Expect(vpcID).To(Equal(*vpcOutput.ID))
+	})
+	t.Run("When Spec VPC ID is not set and GetVPCByName returns success", func(t *testing.T) {
```

Comment:
```suggestion
	t.Run("When spec.VPC.ID is not set and GetVPCByName returns success", func(t *testing.T) {
```
Please change wherever applicable